cvmfs/test/repository_test.py

```diff
@@ -95,3 +95,34 @@ def test_expired_whitelist(self):
         self.assertRaises(cvmfs.RepositoryVerificationFailed,
                           cvmfs.open_repository,
                           self.mock_repo.dir, public_key=self.mock_repo.public_key)
+
+
+    def test_lookup(self):
+        self.mock_repo.make_valid_whitelist()
+        self.mock_repo.serve_via_http()
+        repo = cvmfs.open_repository(self.mock_repo.url,
+                                     public_key=self.mock_repo.public_key)
+        dirent = repo.lookup('/.cvmfsdirtab')
+        self.assertIsNotNone(dirent)
+        dirent = repo.lookup('/bar/4/foo')
+        self.assertIsNotNone(dirent)
+        dirent = repo.lookup('/bar/4/foobar')
+        self.assertIsNone(dirent)
```

Comment:
A check for a non-existing directory entry would be nice. What about relative paths? Paths with a trailing slash, maybe?